### PR TITLE
fix(LIVE-9486): add specific featureFlags deeplinks tutorial

### PR DESCRIPTION
**Pull request by stephane-lieumont-ledger**
### 📝 Description

- Add specific featureflags deeplinks during LLD Onboarding with LNX to redirect on recover.
- Add Custom uri on hook useProtect useCustomPath
- Adjust links to recover Restore and Upsell screen with function useCustomPath
- Fix black screen at the end of onboarding Stax during Recover Live App redirection

### ❓ Context

- **Impacted projects**: `ledger-live-desktop`, `live-common`, `types-live`  
- **Linked resource(s)**: [LIVE-9486](https://ledgerhq.atlassian.net/browse/LIVE-9486) - [PROTECT-2575](https://ledgerhq.atlassian.net/browse/PROTECT-2575) 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-9486]: https://ledgerhq.atlassian.net/browse/LIVE-9486?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PROTECT-2575]: https://ledgerhq.atlassian.net/browse/PROTECT-2575?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/components/Onboarding/Screens/Tutorial/index.tsx [hunk 1]
```diff
@@ -567,15 +571,29 @@ export default function Tutorial({ useCase }: Props) {
...
             fallbackIfNoAction: () => history.push("/"),
           });
-        if (upsellPath) {
+
+        if (useCase === UseCase.setupDevice && upsellPath) {
           history.push(upsellPath);
+        } else if (useCase === UseCase.recoveryPhrase && postOnboardingPath) {
+          history.push(postOnboardingPath);
+        } else if (useCase === UseCase.connectDevice && devicePairingPath) {
+          history.push(devicePairingPath);
```
Comment (Justkant) [inline apps/ledger-live-desktop/src/renderer/components/Onboarding/Screens/Tutorial/index.tsx:580]: Was that added to move to different "route" if you end the onboarding in different screens ?

File: libs/ledger-live-common/src/hooks/recoverFeatureFlag.ts [hunk 2]
```diff
@@ -94,3 +94,51 @@ export function useLoginPath(
...
+export function useCustomPath(
+  servicesConfig: Feature_ProtectServicesDesktop | Feature_ProtectServicesMobile | null,
+  page?: string,
+  source?: string,
+): string | undefined {
+  const uri = usePostOnboardingURI(servicesConfig);
+  const [basicUri] = uri ? uri.split("?") : [];
+  const search = `?${page ? `redirectTo=${page}` : ""}${page && source ? "&" : ""}${
+    source ? `source=${source}` : ""
+  }`;
```
[the remark this answers is not in the document]
Comment (stephane-lieumont-ledger) [reply libs/ledger-live-common/src/hooks/recoverFeatureFlag.ts]: https://github.com/LedgerHQ/ledger-live/pull/5027/commits/a6b1c4ec3b1ae76945ae265a3a86c19ff1e267d7

File: libs/ledger-live-common/src/hooks/recoverFeatureFlag.ts [hunk 3]
```diff
@@ -94,3 +94,74 @@ export function useLoginPath(
...
+  const modelUri = usePostOnboardingURI(servicesConfig);
+  const [basicUri] = modelUri ? modelUri.split("?") : [];
+  const uri = new URL(basicUri);
+
+  if (page) uri.searchParams.append("redirectTo", page);
+  if (source) uri.searchParams.append("source", source);
+  if (source && deeplinkCampaign) {
+    uri.searchParams.append("ajs_recover_source", source);
+    uri.searchParams.append("ajs_recover_campaign", deeplinkCampaign);
+  }
```
Comment (Justkant) [inline libs/ledger-live-common/src/hooks/recoverFeatureFlag.ts:164]: Not sure about the cost of creating an URL and appending to it but I would memo that part
```suggestion
  const uri = useMemo(() => {
    const [basicUri] = modelUri ? modelUri.split("?") : [];
    const uri = new URL(basicUri);

    if (page) uri.searchParams.append("redirectTo", page);
    if (source) uri.searchParams.append("source", source);
    if (source && deeplinkCampaign) {
      uri.searchParams.append("ajs_recover_source", source);
      uri.searchParams.append("ajs_recover_campaign", deeplinkCampaign);
    }
    return uri;
  }, [deeplinkCampaign, modelUri, page, source]);
```

**Review by Justkant — CHANGES_REQUESTED** [review]
Small change I would do

File: libs/ledger-live-common/src/hooks/recoverFeatureFlag.ts (hunk 3 above)
Comment (stephane-lieumont-ledger) [reply libs/ledger-live-common/src/hooks/recoverFeatureFlag.ts]: https://github.com/LedgerHQ/ledger-live/pull/5027/commits/843834a03d7419b1708b32efce6f1deb5cc3b701

**Review by Justkant — APPROVED** [review]

**Pull request merged**